api/src/main/java/org/pmiops/workbench/institution/InstitutionServiceImpl.java

```diff
@@ -206,25 +215,37 @@ public boolean validateInstitutionalEmail(Institution institution, String contac
       return false;
     }
 
-    // If the Institution has DUA Agreement that is restricted just to few researchers
-    // Confirm if the email address is in the allowed email list
-    if (institution.getDuaTypeEnum() != null
-        && institution.getDuaTypeEnum().equals(DuaType.RESTRICTED)) {
-      final boolean validated = institution.getEmailAddresses().contains(contactEmail);
+    // As of now, RT's short name is hard coded in AccessTierService. We may need a better way
+    // to pull RT short name from config or database.
+    Optional<InstitutionTierRequirement> rtRequirement =
+        getTierRequirement(institution, REGISTERED_TIER_SHORT_NAME);
+
+    // If the Institution's registered tier agreement requires email addresses confirm if the email
+    // address is in the allowed email list.
+    if (rtRequirement.isPresent()
```

Comment:
Trying to understand this `isPresent()` check. Should this just error out instead if this is not present? Why continue to checking domains?

Reply:
That is to similar to the previous DuaType version. If DuaType is null, treat it the same as MASTER. 
```
if DuaType == ADDRESSES {
  checkAddress
} else {
 checkDomains
}
```

But I guess I can remove this, and assume RT requirement will never be null.